src/Illuminate/Database/Eloquent/Casts/ArrayObject.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Illuminate\Database\Eloquent\Casts;
+
+use ArrayObject as BaseArrayObject;
+use Illuminate\Contracts\Support\Arrayable;
+use JsonSerializable;
+
+class ArrayObject extends BaseArrayObject implements Arrayable, JsonSerializable
+{
+    /**
+     * Get a collection containing the underlying array.
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    public function collect()
+    {
+        return collect($this->getArrayCopy());
```

Comment:
Because the `ArrayObject` is already `Traversable`, you don't need to get an `->getArrayCopy()` here.

Reply:
I believe one of my project is exactly waiting for that